src/data/repositories/TrainingModuleDefaultRepository.ts

```diff
@@ -21,24 +21,54 @@ import {
     PersistedTrainingModule,
     TranslationConnection,
 } from "../entities/PersistedTrainingModule";
+//added
+import { D2Api } from "../../types/d2-api";
+import { getD2APiFromInstance } from "../utils/d2-api";
+
+interface ModuleResponse {
+    data: string;
+    headers: Headers;
+    status: number;
+}
+
+interface Headers {
+    cacheControl: string;
+    contentType: string;
+    etag: string;
+    lastModified: Date;
+}
 
 export class TrainingModuleDefaultRepository implements TrainingModuleRepository {
     private builtinModules: Dictionary<JSONTrainingModule | undefined>;
     private storageClient: StorageClient;
     private progressStorageClient: StorageClient;
+    //added
+    private api: D2Api;
 
     constructor(private config: ConfigRepository) {
         this.builtinModules = BuiltinModules;
         this.storageClient = new DataStoreStorageClient("global", config.getInstance());
         this.progressStorageClient = new DataStoreStorageClient("user", config.getInstance());
+        //added
+        this.api = getD2APiFromInstance(config.getInstance());
     }
-
     public async list(): Promise<TrainingModule[]> {
         try {
             const dataStoreModules = await this.storageClient.listObjectsInCollection<
                 PersistedTrainingModule
             >(Namespaces.TRAINING_MODULES);
 
+            dataStoreModules.forEach(async module => {
+                try{
+                    await this.api.baseConnection.request<ModuleResponse>({ method: "get", url: module.dhisLaunchUrl }).response
+                    .then(() =>  module.installed = true);
+                }
+                catch(e) {
+                    console.log(e.response);
+                    module.installed = false;
+                }
+            });
```

Comment:
A more functional approach would use a ``dataStoreModules.map`` since it is async we would normally use ``promiseMap(dataStoreModules, async () => {})``